pinot-core/src/test/java/org/apache/pinot/queries/JsonDataTypeQueriesTest.java

```diff
@@ -302,6 +305,54 @@ public void testJsonMatchArrayWithIndex() {
     Assert.assertEquals(iterator.next()[0], "goofy");
   }
 
+  @Test
+  public void testJsonMatchNestedArrayWithIndex() {
```

Comment:
(nit)

Test case naming is confusing. Both are testing multi-dimensional array IMO

data is an array, e is an array inside data array and y is an array inside e array

Reply:
If you have a good suggestion for name, we can use that, otherwise can leave it as it is?

Reply:
Agreeing on the terminology is more important. So my confusion is that nested array and MD array are same things?
Essentially an array inside an array. each element is an array data -> e -> y 
It's ok to leave the test case name as is but logically what we use in docs/design etc should be unified I think